sdk/storage/Azure.Storage.Common/tests/Shared/TestEventListener.cs

```diff
@@ -19,23 +20,45 @@ namespace Azure.Storage.Test
     /// Simply create an instance of the TestEventListener before you start
     /// running your tests.
     /// </summary>
-    internal class TestEventListener : AzureEventSourceListener
+    internal class TestEventListener : IDisposable
     {
-        public TestEventListener() : base((e, _) => LogEvent(e), EventLevel.Verbose)
+        private StringBuilder _eventBuffer;
+
+        private readonly AzureEventSourceListener _eventSourceListener;
+
+        public TestEventListener()
         {
+            _eventSourceListener = new AzureEventSourceListener(
+                (e, _) => LogEvent(e),
+                EventLevel.Verbose);
         }
 
         /// <summary>
-        /// Trace any SDK events.
+        /// Sets up the Event listener buffer for the test about to run.
         /// </summary>
-        /// <param name="args">Event arguments.</param>
-        public static void LogEvent(EventWrittenEventArgs args)
+        public void SetupEventsForTest()
+        {
+            _eventBuffer = new StringBuilder();
+        }
+
+        /// <summary>
+        /// Output the Events to the console in the case of test failure.
+        /// This will include the HTTP requests and responses.
+        /// </summary>
+        public void OutputEventsForTest()
         {
-            if (!Debugger.IsAttached)
+            if (TestContext.CurrentContext.Result.Outcome.Status == TestStatus.Failed)
             {
-                return;
+                TestContext.Progress.WriteLine(_eventBuffer.ToString());
```

Comment:
Why TestContext.Progress instead of TestContext.Error?

Reply:
I think Error should work. Progress is a holdover from when I was outputting events for all tests. I will need to test it out though to make sure it works as expected.

Reply:
Error ends up printing the logs twice. It seems to get printed immediately and then again as part of the error information.